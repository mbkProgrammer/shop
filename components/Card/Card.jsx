import styled, { Styled } from '@emotion/styled';
import { useRouter } from 'next/router';

const Card = ({
  imageSrc = './assets/img/not-found.png',
  imageAlt = 'image',
  path = '/',
  price = '0.0',
}) => {
  const router = useRouter();
  const Card = styled.div`
    margin: 10px;
    width: 250px;
    height: 230px;
    display: flex;
    flex-direction: column;
    border-radius: 5px;
    box-shadow: 5px 10px 3px #999;
    border: 2px solid var(--primary);
    overflow: hidden;
    @media (max-width: 480px) {
      box-shadow: 0px 2px 3px #999;
      margin: auto;
      flex-direction: row;
      width: 95vw;
      height: 30vw;
      margin-bottom: 3px;
    }
  `;
  const CardImg = styled.img`
    height: 160px;
    border: none;
    border-radius: 5px;
    @media (max-width: 480px) {
      height: 100%;
      width: 45%;
    }
  `;

  const CardContent = styled.div`
    display: flex;
    flex-direction: column;
    overflow: hidden;
    justify-content: space-between;
  `;

  const CardTittle = styled.h4`
    font-weight: 500;
    line-height: 30px;
    font-size: 16px;
    margin-left: 5px;
    color: rgba(29, 41, 63, 1);
  `;
  const CardPrice = styled.h4`
    align-self: end;
    padding: 10px;
    font-weight: 500;
    font-size: 18px;
    line-height: 18px;
    text-align: right;
    color: var(--primary);
    &::after{
      content: '  $';
      color: var(--primary);
  }
  `;

  return (
    <Card onClick={() => router.push(path)}>
      <CardImg src={imageSrc} alt={imageAlt} />
      <CardContent>
        <CardTittle>
          lorem10Sit eu nulla incididunt velit ea anim fugiat .
        </CardTittle>
        <CardPrice>{price}</CardPrice>
      </CardContent>
    </Card>
  );
};

export default Card;
