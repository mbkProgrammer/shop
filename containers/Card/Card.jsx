import styled, { Styled } from '@emotion/styled';
import { useRouter } from 'next/router';

const Card = ({
  imageSrc = './assets/img/not-found.png',
  imageAlt = 'image',
  path = '/',
  title = 'this is title himmdcs csdcscd ',
  price = '0.0',
}) => {
  const router = useRouter();
  const Card = styled.div`
  position: relative;
    margin: 15px;
    width: 250px;
    height: 230px;
    display: flex;
    flex-direction: column;
    border-radius: 5px;
    box-shadow: 5px 10px 3px #999;
    border: 2px solid var(--primary);
    overflow: hidden;
    transition: 0.4s;
    &:hover {
      transform: translateY(-10px) scale(1.01);
    }
    @media (max-width: 570px) {
      box-shadow: 0px 2px 3px #999;
      margin: auto;
      flex-direction: row;
      width: 95vw;
      height: 25vw;
      margin-bottom: 5px;
      &:hover {
        transform: scale(1.01);
      }
    }
  `;
  const CardImg = styled.img`
    height: 150px;
    width: 250px;
    border: none;
    border-radius: 5px;
    overflow: hidden;
    @media (max-width: 570px) {
      height: 100%;
      width: 65vw;
    }
  `;

  const CardContent = styled.div`
    display: flex;
    flex-direction: column;
    width: 100%;
    justify-content: space-between;
  `;

  const CardTittle = styled.h4`
    font-weight: 500;
    line-height: 22px;
    font-size: 16px;
    margin-left: 5px;
    height: 100%;
    overflow: hidden;
    color: rgba(29, 41, 63, 1);
    @media (max-width: 300px){
      font-size: 14px;
    }
  `;
  const CardPrice = styled.h4`
    align-self: end;
    height: 32px;
    font-weight: 500;
    font-size: 18px;
    line-height: 18px;
    text-align: center;
    padding-top: 10px;
    padding-right: 5px;
    display: block;
    position: absolute;
    bottom: 2px;
    right: 2px;
    color: var(--primary);
    &::after{
      content: '  $';
      color: var(--primary);
  }
  `;

  return (
    <Card onClick={() => router.push(path)}>
      <CardImg src={imageSrc} alt={imageAlt} />
      <CardContent>
        <CardTittle>
          {title}
        </CardTittle>
        <CardPrice>{price}</CardPrice>
      </CardContent>
    </Card>
  );
};

export default Card;
