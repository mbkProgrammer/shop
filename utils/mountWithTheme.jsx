import { ThemeProvider } from '@emotion/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../configs/Store';
import theme from '../configs/theme';

const mountWithTheme = (children) => renderer.create(
  <Provider store={store}>
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  </Provider>,
);

export default mountWithTheme;
