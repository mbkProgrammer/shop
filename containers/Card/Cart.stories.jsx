import React from 'react';
import Card from './Card';

export default { title: 'containers/Card' };
export const CardTest = () => (<Card imageSrc="https://picsum.photos/250/150.jpg" imageAlt="test" price="10.0" title="this is tittle" path="/write-path-when-onclick" />);
