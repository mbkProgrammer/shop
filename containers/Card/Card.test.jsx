import mountWithTheme from '../../utils/mountWithTheme';
import Card from './Card';

test('Card sample', () => {
  const component = mountWithTheme(
    <Card imageAlt="test" imageSrc="" path="./" price="10.0" title="test" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
