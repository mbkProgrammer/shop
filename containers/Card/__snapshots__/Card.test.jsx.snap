// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card sample 1`] = `
<div
  className="css-cpn2qo"
  onClick={[Function]}
>
  <img
    alt="test"
    className="css-coeihp"
    src=""
  />
  <div
    className="css-1rs1nzp"
  >
    <h4
      className="css-jbfoo9"
    >
      test
    </h4>
    <h4
      className="css-4m54hb"
    >
      10.0
    </h4>
  </div>
</div>
`;
